project/npda/general_functions/csv/csv_upload.py

```diff
@@ -305,33 +322,25 @@ async def process_rows_for_patient(rows, async_client):
             # Pull through cleaned_data so we can use it in the async visit validators
             await sync_to_async(patient_form.is_valid)()
 
-            record_errors_from_form(errors_to_return, patient_row_index, patient_form)
-
             visit_forms = []
             for _, row in rows.iterrows():
                 visit_form = await validate_visit_using_form(
                     patient_form, row, async_client
                 )
-                visit_forms.append((visit_form, int(row["row_index"])))
+                
+                # Pull through cleaned_data
+                visit_form.is_valid()
 
-            nhs_number = patient_form.cleaned_data.get("nhs_number")
```

Comment:
We already had errors in the form but now I've rearranged the code we weren't saving them - just the special case error that was in `process_rows_for_patient`.

I've moved the guard from the special case into `save_patient_and_transfer` so that we have an opportunity to gather the existing errors first.

This means we can remove the redundant `__all__` error in favour of the existing errors on both fields that already existed from the form.